client/src/components/History/SwitchToHistoryLink.test.ts

```diff
@@ -94,7 +125,56 @@ describe("SwitchToHistoryLink", () => {
             deleted: false,
             purged: false,
             archived: true,
-        } as HistorySummary;
+            user_id: "user_id",
+        } as HistorySummaryExtended;
         await expectOptionForHistory("View", history);
     });
+
+    it("should view in new tab when the history is unowned and published", async () => {
+        const history = {
+            id: "unowned-history-id",
+            name: "History Published",
+            deleted: false,
+            purged: false,
+            archived: false,
+            published: true,
+            user_id: "other_user_id",
+        } as HistorySummaryExtended;
+        const wrapper = await expectOptionForHistory("View", history);
+
+        // Wait for the history to be loaded
+        await flushPromises();
+
+        // Click on the history link
+        const link = wrapper.find(selectors.historyLinkClick);
+        await link.trigger("click");
+        await wrapper.vm.$nextTick();
+
+        // History opens in a new tab
+        expect(Toast.error).toHaveBeenCalledTimes(0);
+    });
+
+    it("should display View option and show error on click when history is unowned and not published", async () => {
```

Comment:
```suggestion
    it("should display View option and show error on click when history is not accessible", async () => {
```
Should it not just be disabled with a tooltip ? In fact this seems like an impossible scenario, as you can't fetch an inaccessible history from a server (unless you're admin, but then you can access everything anyway) ?